packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte

```diff
@@ -0,0 +1,190 @@
+import WebAssemblyIcon from '../images/WebAssemblyIcon.svelte';
+import { onMount } from 'svelte';
+
+export let platforms: string = '';
+let platformsArray: string[] = [];
+
+interface CardInfo {
+  title: string;
+  badge: string;
+  isDefault: boolean;
+  checked: boolean;
+  value: string;
+  icon: unknown;
+  fontAwesomeIcon: boolean;
+}
+
+const DEFAULT_CARDS: CardInfo[] = [
+  {
+    title: 'Linux',
+    badge: 'AMD64',
+    isDefault: false,
+    checked: false,
+    value: 'linux/amd64',
+    icon: faLinux,
+    fontAwesomeIcon: true,
+  },
+  {
+    title: 'Linux',
+    badge: 'ARM64',
+    isDefault: false,
+    checked: false,
+    value: 'linux/arm64',
+    icon: faLinux,
+    fontAwesomeIcon: true,
+  },
+];
+
+const ADVANCED_CARDS: CardInfo[] = [
+  {
+    title: 'Linux',
+    badge: 'PPC64LE',
+    isDefault: false,
+    checked: false,
+    value: 'linux/ppc64le',
+    icon: faLinux,
+    fontAwesomeIcon: true,
+  },
+  {
+    title: 'Linux',
+    badge: 'S390X',
+    isDefault: false,
+    checked: false,
+    value: 'linux/s390x',
+    icon: faLinux,
+    fontAwesomeIcon: true,
+  },
+  {
+    title: 'WebAssembly',
+    badge: 'WASI',
+    isDefault: false,
+    checked: false,
+    value: 'wasi/wasm',
+    icon: WebAssemblyIcon,
+    fontAwesomeIcon: false,
+  },
+];
+
+let showMoreOptions = false;
+
+let sortedCards: CardInfo[] = [];
+let advancedCards: CardInfo[] = [];
+
+onMount(async () => {
+  // need to sort the default cards by having the first card using the same arch than Podman Desktop
+
+  // get current arch
+  let arch = await window.getOsArch();
+  if (arch === 'x64') {
+    arch = 'amd64';
+  }
+
+  // sort the default cards using current arch as first
+  DEFAULT_CARDS.sort((a, b) => {
+    if (a.value.includes(arch)) {
+      return -1;
+    }
+    if (b.value.includes(arch)) {
+      return 1;
+    }
+    return 0;
+  });
+  sortedCards = DEFAULT_CARDS;
+
+  // first item should be the default
+  sortedCards[0].isDefault = true;
+  sortedCards[0].checked = true;
+
+  advancedCards = ADVANCED_CARDS;
+});
+
+function handleCard(card: { detail: { mode: 'add' | 'remove'; value: string } }) {
+  if (card.detail.mode === 'add') {
+    // add card.detail.value from the array platformsArray
+    platformsArray.push(card.detail.value);
+  } else if (card.detail.mode === 'remove') {
+    // remove card.detail.value from the array platformsArray
+    platformsArray = platformsArray.filter(item => item !== card.detail.value);
+  }
+
+  // platforms should be a comma separated string
+  platforms = platformsArray.join(',');
+}
+
+function addCard(item: { value: string }) {
+  // create a new card and make it checked
+  const card = {
+    title: item.value,
+    badge: 'custom',
+    isDefault: false,
+    checked: true,
+    value: item.value,
+    icon: faLayerGroup,
+    fontAwesomeIcon: true,
+  };
+  advancedCards.push(card);
+  advancedCards = advancedCards;
+}
+</script>
+
+<div class="flex flex-col">
```

Comment:
I would consider also adding ARIA functionality from the start of this PR. 
adding `role="region" aria-label="Build Platform Options"` or something similar would be nice.

Reply:
added